### ISSUE TITLE
Questions about training.

### ISSUE THREAD

**Issue by smallccn** [FreedomIntelligence/HuatuoGPT-o1#23]
I tried to reproduce the training of stage1 and stage2 in my local environment. 

I used **Qwen2.5-7B-Instruct** as the base model and used **medical-o1-reasoning-SFT** and **medical-o1-verifiable-problem** as the training data. 

Strangely, after completing the training of stage1 and stage2 and obtaining a fine-tuned model, the **accuracy** of my fine-tuned model was lower than that of the original **Qwen2.5-7B-Instruct** model when evaluated on the test data included in the project. However, the accuracy of [HuatuoGPT-o1-7B](https://huggingface.co/FreedomIntelligence/HuatuoGPT-o1-7B) was very high. 

I'm not sure where the problem lies. Can anyone offer some advice?

**Comment by DIaacKr** [comment]
It also confused me a lot... 
Is there anyone who can explain it?

**Comment by ytytbyt** [comment]
> I tried to reproduce the training of stage1 and stage2 in my local environment.
> 
> I used **Qwen2.5-7B-Instruct** as the base model and used **medical-o1-reasoning-SFT** and **medical-o1-verifiable-problem** as the training data.
> 
> Strangely, after completing the training of stage1 and stage2 and obtaining a fine-tuned model, the **accuracy** of my fine-tuned model was lower than that of the original **Qwen2.5-7B-Instruct** model when evaluated on the test data included in the project. However, the accuracy of [HuatuoGPT-o1-7B](https://huggingface.co/FreedomIntelligence/HuatuoGPT-o1-7B) was very high.
> 
> I'm not sure where the problem lies. Can anyone offer some advice?

The paper mentions adding 4K unconverted data and 5K general verification questions. Are both the 4K unconverted data and the 5K general verification questions only used during the RL training stage?

**Comment by qq869393162** [comment]
> > 我试图在当地环境中重现第 1 阶段和第 2 阶段的训练。
> > 我使用 **Qwen2.5-7B-Instruct** 作为基础模型，并使用 **medical-o1-reasoning-SFT** 和 **medical-o1-verifiable-problem** 作为训练数据。
> > 奇怪的是，在完成 stage1 和 stage2 的训练并获得微调模型后，在项目中包含的测试数据上评估时，我的微调模型的**准确率**低于原始 **Qwen2.5-7B-Instruct** 模型。然而，[华佗GPT-o1-7B](https://huggingface.co/FreedomIntelligence/HuatuoGPT-o1-7B)的准确率非常高。
> > 我不确定问题出在哪里。谁能提供一些建议？
> 
> 论文提到增加了 4K 未转换数据和 5K 一般验证问题。4K 未转换数据和 5K 通用验证题是否都只在 RL 训练阶段使用？

想请问一下，4K 未转换数据和 5K 一般验证问题论文是不是并未开源，我这边没找到

**Comment by ytytbyt** [comment]
是的没有开源， 关于数据集的问题我发邮件问了作者但没回复

qq869393162 ***@***.***> 于2025年8月12日周二 16:11写道：

**Comment by qq869393162** [comment]
已收到你的资料！（重发记得备注好）

**Comment by qq869393162** [comment]
> 是的没有开源， 关于数据集的问题我发邮件问了作者但没回复
> 
> qq869393162 ***@***.***> 于2025年8月12日周二 16：11写道：
> […](#)

想请问下你的复现实验做得怎么样？我这边的avg.跟HuatuoGPT-o1-8B的模型差了1个来点，做了部分实验发现RL的效果提升并不是很明显。方便交流一下吗

**Comment by ytytbyt** [comment]
> > 是的没有开源， 关于数据集的问题我发邮件问了作者但没回复
> > qq869393162 _**@**_.***> 于2025年8月12日周二 16：11写道：
> > […](#)
> 
> 想请问下你的复现实验做得怎么样？我这边的avg.跟HuatuoGPT-o1-8B的模型差了1个来点，做了部分实验发现RL的效果提升并不是很明显。方便交流一下吗

stage1做了，stage2还没做；结果有一点点出入应该也算正常吧

**Comment by RXH04-USTC** [comment]
我复现的stage1的结果和原文差距也比较大

<img width="550" height="1238" alt="Image" src="https://github.com/user-attachments/assets/b4dfa903-b2b1-47be-b334-509d19bbf8d3" />

**Comment by qq869393162** [comment]
stage1我倒是没问题，我比paper好一两个点。



发自我的iPhone


------------------ Original ------------------
From: renxiaohan ***@***.***&gt;
Date: Tue,Sep 16,2025 4:16 PM
To: FreedomIntelligence/HuatuoGPT-o1 ***@***.***&gt;
Cc: Zhong ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [FreedomIntelligence/HuatuoGPT-o1] Questions about training.(Issue #23)



renxiaohan666 left a comment (FreedomIntelligence/HuatuoGPT-o1#23)
 
我复现的stage1的结果和原文差距也比较大
 image.png (view on web)
